src/App.js

```diff
@@ -152,29 +154,154 @@ class Board extends React.Component {
     }
   }
 
+  /**
+   * This method checks the solutions for the specified row.
+   * A row is valid if there's 9 values and all numbers 1-9 are present.
+   *
+   * @returns boolean, true if valid row, false otherwise
+   * @param {*} rowToVerify, an integer number
+   * @param {*} board, board is a 1D array representing Sudoku board
+   */
+  static isRowComplete(rowToVerify, board) {
+    let numCol = 9;
+    let maxSudokuValue = 9;
+    const rowSet = new Set(); //create new set to store values of row
+    
+    //add values of board into Set
+    for (let col = 0; col < numCol; col++) {
+      //board is a 1D array, so we need to get the element num
+      rowSet.add(board[(9 * rowToVerify) + col]); 
+    }
+    //should have 9 values in row
+    if (rowSet.size !== maxSudokuValue) {
+      return false;
+    }
+
+    // check if all possible values 1-9 are all in rowSet
+    for (let possibleSudoKuValue = 1; possibleSudoKuValue <= maxSudokuValue; possibleSudoKuValue++) {
+      if (rowSet.has(possibleSudoKuValue) === false) {
+        return false;
+      }
+    }
+    return true; //we've check that all numbers 1-9 are in the row
```

Comment:
I'm wondering if the === will make a difference? It looks like JS has autoboxing/auto unboxing, so I'm not sure why it always returns false

Reply:
No, rowSet.has(possibleSudoKuValue) is always false. If you log it, you'll see it